config/v1/types_infrastructure.go

```diff
@@ -28,6 +28,11 @@ type InfrastructureSpec struct {
 	// The namespace for this config map is openshift-config.
 	// +optional
 	CloudConfig ConfigMapFileReference `json:"cloudConfig"`
+
+	// platformSpec holds desired information specific to the underlying
+	// infrastructure provider.
+	// +optional
+	PlatformSpec *PlatformSpec `json:"platformSpec,omitempty"`
```

Comment:
I think an empty type implies missing.  We try to avoid pointers.

Reply:
since go marshal/unmarshal doesn't omit empty structs, I think pointer is a better choice here.

Also this is important for migrations as detailed in https://github.com/openshift/api/pull/599/files#r401189373

Reply:
> since go marshal/unmarshal doesn't omit empty structs, I think pointer is a better choice here.

We are ok with including empty values in serialization. In some ways it's even good since it outlines what is available and what is not.  We've been avoiding pointer in our config structs to good effect.  If migration cannot be handled another way we would do it, but I don't think that's the case today.

By allowing empty string in Type, you can distinguish intent.